quickwit/quickwit-indexing/src/split_store/indexing_split_store.rs

```diff
@@ -201,34 +88,30 @@ impl IndexingSplitStore {
     /// Creates an instance of [`IndexingSplitStore`]
     ///
     /// It needs the remote storage to work with.
-    pub async fn create_with_local_store(
+    pub fn new(
         remote_storage: Arc<dyn Storage>,
-        cache_directory: &Path,
         merge_policy: Arc<dyn MergePolicy>,
-        split_store_space_quota: Arc<Mutex<SplitStoreSpaceQuota>>,
-    ) -> StorageResult<Self> {
-        let local_storage_root = cache_directory.join(SPLIT_CACHE_DIR_NAME);
-        std::fs::create_dir_all(&local_storage_root)?;
-        let local_split_store =
-            LocalSplitStore::open(local_storage_root, split_store_space_quota).await?;
-
+        local_split_store: Arc<LocalSplitStore>,
+    ) -> Self {
         let inner = InnerIndexingSplitStore {
             remote_storage,
-            local_split_store: Some(Mutex::new(local_split_store)),
+            local_split_store,
             merge_policy,
         };
-        Ok(Self {
+        Self {
             inner: Arc::new(inner),
-        })
+        }
     }
 
+    /// Helper function to create a indexing split store for tests.
+    /// The resulting store does not have any local cache.
```

Comment:
`:#[cfg(any(test, feature = "testsuite"))]`?

Reply:
we can't really do this because it is also used in the janitor.

Reply:
So let's change the comment t_t